.github/workflows/upload-docx.yml

```diff
@@ -0,0 +1,103 @@
+name: Generate DOCX
+
+on:
+  push:
+    branches:
+      - 'latest'
+      - '3.0'
+
+jobs:
+  generate-singlehtml:
+    runs-on: ubuntu-latest
+    container: tarantool/doc-builder:fat-4.3
+    steps:
+      - uses: actions/checkout@v3
+        id: checkout
+        with:
+          submodules: recursive
+
+      - name: generate singlehtml
+        run: |
+          cmake .
+          make pull-modules
+          make build-modules
+          make singlehtml
+          make singlehtml-ru
+
+      - name: save singlehtml-en
+        uses: actions/upload-artifact@v3
+        with:
+          name: singlehtml-en
+          path: output/html/en/singlehtml.html
+
+      - name: save singlehtml-ru
+        uses: actions/upload-artifact@v3
+        with:
+          name: singlehtml-ru
+          path: output/html/ru/singlehtml.html
+
+      - name: save pandoc filters from the repo
+        uses: actions/upload-artifact@v3
+        with:
+          name: pandoc
+          path: pandoc/filter-espd.py
+  
+  generate-docx:
+    needs: generate-singlehtml
+    runs-on: ubuntu-latest
+    container: python:3.8-slim-bookworm
+    env:
+      AWS_ACCESS_KEY_ID: ${{secrets.AWS_ACCESS_KEY_ID}}
+      AWS_SECRET_ACCESS_KEY: ${{secrets.AWS_SECRET_ACCESS_KEY}}
+      AWS_DEFAULT_REGION: ${{secrets.AWS_DEFAULT_REGION}}
+      S3_ENDPOINT_URL: ${{secrets.S3_ENDPOINT_URL}}
+      S3_UPLOAD_PATH: ${{secrets.S3_DOCX_PATH}}
+    steps:
+      - name: install dependencies
```

Comment:
It seems to me that this should be done once (create and save an image with dependencies), and then use the already image in our workflow.